### PR TITLE
Fix flickering icons on sitemap reload.

### LINKED ISSUE #541: Icons flash when sitemap is reloaded
Navigate to any page in your sitemap and wait until some item changes. All icons in the sitemap flash.

Reported here: https://community.openhab.org/t/android-and-ios-2-0-releases/37405/3?u=mueller-ma

### PR DESCRIPTION
**Pull request by maniac103**
Fixes #541

### CODE REVIEW

**Comment by mueller-ma** [comment]
LGTM!

**Comment by mueller-ma** [comment]
@maniac103 Can you sign your commit or ask me to sign it for you, when I merge this PR?

**Comment by maniac103** [comment]
I'll fix it. Sorry that I keep forgetting it. I wonder whether there's a per-repo option in git to do it automatically.

**Comment by mueller-ma** [comment]
Do you commit via cli? Bash alias is your friend ;)

**Comment by maniac103** [comment]
> I wonder whether there's **a per-repo option** in git

;-)

https://stackoverflow.com/questions/15015894/git-add-signed-off-by-line-using-format-signoff-not-working should do it though

File: mobile/src/main/java/org/openhab/habdroid/util/MySmartImageView.java [hunk 1]
```diff
@@ -95,35 +96,47 @@ public MySmartImageView(Context context, AttributeSet attrs, int defStyle) {
...
+        setImageUrl(url, username, password, useImageCache, null);
     }
 
-    public void setImageUrl(String url, final Integer fallbackResource, final Integer loadingResource, String username, String password) {
-        this.myImageUrl = url;
-        this.username = username;
-        this.password = password;
-        this.refreshDisabled = true;
-        setImage(new MyWebImage(url, username, password), fallbackResource, loadingResource, imageCompletionListener);
+    public void setImageUrl(String url, String username, String password, Integer fallbackResource) {
```
[the remark this answers is not in the document]
Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/util/MySmartImageView.java]: @maniac103 New line because of one char too much?

File: mobile/src/main/java/org/openhab/habdroid/ui/drawer/OpenHABDrawerAdapter.java [hunk 2]
```diff
@@ -98,8 +98,7 @@ public View getView(int position, View convertView, ViewGroup parent) {
                 }
                 if (siteMap.getIcon() != null && drawerItemImage != null) {
                     String iconUrl = openHABBaseUrl + Uri.encode(siteMap.getIconPath(),"/?=");
-                    drawerItemImage.setImageUrl(iconUrl, R.mipmap.icon,
-                            openHABUsername, openHABPassword);
+                    drawerItemImage.setImageUrl(iconUrl, openHABUsername, openHABPassword, R.mipmap.icon);
```
[the remark this answers is not in the document]
Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/ui/drawer/OpenHABDrawerAdapter.java]: @maniac103

**Comment by mueller-ma** [comment]
Some of the stickler errors are actually valid

**Comment by maniac103** [comment]
That may be true, but it's hard to follow code style if large parts of the existing code base don't follow it.
I can update my changes to fully conform, but I wonder whether it wouldn't be a better idea to turn off Stickler for now, merge the large, invasive PRs and afterwards do a cleanup of the code base first: fix whitespace, name variables consistently, remove unused code etc. ... then enable Stickler again afterwards. I volunteer to help with the cleanup, we just need to coordinate when to do it as PR rebase will become painful afterwards.
